packages/components/src/templates/next/components/complex/Image/Image.tsx

```diff
@@ -1,26 +1,88 @@
+import type { PropsWithChildren } from "react"
+
 import type { ImageProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
+import { isExternalUrl } from "~/utils"
+import { Link } from "../../internal/Link"
+
+const createImageStyles = tv({
+  slots: {
+    container: "mt-0 [&:not(:first-child)]:mt-7",
+    caption:
+      "overflow-wrap break-word prose-label-sm-medium mt-2 max-w-[100ch] text-base-content-subtle lg:mx-auto lg:text-center",
+    image: "mx-auto h-auto max-w-full rounded",
+  },
+  variants: {
+    size: {
+      small: {
+        image: "min-w-full xs:min-w-[67%] md:min-w-[33%] lg:min-w-0",
+      },
+      medium: {
+        image: "min-w-full xs:min-w-[33%] md:min-w-0",
+      },
+      large: { image: "min-w-0" },
+    },
+  },
+})
+const compoundStyles = createImageStyles()
 
-const BaseImage = ({ src, alt, width }: Omit<ImageProps, "type">) => (
-  <img src={src} alt={alt} width={`${width ?? 100}%`} height="auto" />
+// Enforce a certain minimum width of image if screen size is small
+const getWidthSize = (width: ImageProps["width"]) => {
+  if (width === undefined || width >= 66) {
+    return "large"
+  } else if (width >= 40) {
+    return "medium"
+  } else {
+    return "small"
+  }
+}
+
+const ImageContainer = ({
+  href,
+  LinkComponent,
+  children,
+}: PropsWithChildren<Pick<ImageProps, "href" | "LinkComponent">>) => (
+  <div className={compoundStyles.container()}>
+    {href !== undefined ? (
+      <Link
+        href={href}
+        target={isExternalUrl(href) ? "_blank" : undefined}
+        rel={isExternalUrl(href) ? "noopener noreferrer nofollow" : undefined}
+        LinkComponent={LinkComponent}
+      >
+        {children}
+      </Link>
+    ) : (
+      <>{children}</>
+    )}
+  </div>
 )
 
-const Image = ({ src, alt, width, href }: ImageProps) => {
+const Image = ({
+  src,
+  alt,
+  caption,
+  width,
+  href,
+  assetsBaseUrl,
+  LinkComponent,
+}: ImageProps) => {
   return (
-    <div>
-      {href ? (
-        <a
-          href={href}
-          target={href.startsWith("http") ? "_blank" : undefined}
-          rel={
-            href.startsWith("http") ? "noopener noreferrer nofollow" : undefined
-          }
-        >
-          <BaseImage src={src} alt={alt} width={width}></BaseImage>
-        </a>
-      ) : (
-        <BaseImage src={src} alt={alt} width={width}></BaseImage>
-      )}
-    </div>
+    <ImageContainer href={href} LinkComponent={LinkComponent}>
+      <img
+        src={isExternalUrl(src) ? src : `${assetsBaseUrl}${src}`}
```

Comment:
how are you asserting assetsBaseUrl is present for all external URLs?

Reply:
Hmm we are not asserting that it must be present (hence the undefined). If it is not present, it is treated as if it is not present and rendered as a relative link which is the existing behaviour.

Reply:
if `assetsBaseUrl` is undefined, string templating will output the `"undefined"` string, which may not be what we want yeah

Reply:
Ah okay, updated in e51400d2b8372bc37cc56426a5e9ebd4d168ee18.